large_image/tilesource/base.py

```diff
@@ -1638,6 +1661,9 @@ def getMetadata(self):
             'magnification': mag['magnification'],
             'mm_x': mag['mm_x'],
             'mm_y': mag['mm_y'],
+            # Use private attribute; public property
+            # incurs calcuation cost if _dtype is still None
+            'dtype': self._dtype,
```

Comment:
We should measure just what the time impact is on our test set (that is, on all the files in the externaldata and test/test_files locations) and decide if the performance hit is significant.

Reply:
For local files, I suspect the performance hit will be marginal. I'm concerned about the performance hit for remote files with GDAL-based sources when going through the [virtual filesystem mechanisms](https://gdal.org/user/virtual_file_systems.html) -- and especially so if the image isn't cloud optimized as then `getMetadata()` could require downloading the entire image

Reply:
I've started a timing test result sheet here: https://docs.google.com/spreadsheets/d/1gY-VMdph4RxI5dsw1i0y5SSaZ8oaAJ0lZZ_Fi97gd9g/edit?usp=sharing (kitware access only)